README.md

```diff
@@ -46,48 +46,71 @@ Have a security issue you want to let us know? You can contact us on our website
 
 Our main website is https://polykey.com
 
+## Flakes
+
+Polykey-CLI uses flakes for its build process. By default Nix does not enable this feature.
+To enable it by default, add the following to `~/.config/nix/nix.conf` or `/etc/nix/nix.conf`:
+
+```
+experimental-features = nix-command flakes
+```
+
+Or if you're on NixOS, you can add it to your flakes system configuration:
+
+```
+nix.settings.experimental-features = [ "nix-command" "flakes" ];
+```
+
+Alternatively, to enable it temporarily append the following to any nix related commands:
+
+```
+--extra-experimental-features flakes
+```
+
 ## Installation
 
 Note that JavaScript libraries are not packaged in Nix. Only JavaScript applications are.
 
 Building the package:
 
 ```sh
-npmDepsHash="$(prefetch-npm-deps ./package-lock.json)"
-nix-build -E "(import ./pkgs.nix {}).callPackage ./default.nix { npmDepsHash = \"$npmDepsHash\"; }"
+nix build .#polykey-cli
 ```
 
 ### Nix/NixOS
 
 Building the releases:
 
 ```sh
-nix-build ./release.nix --attr application --argstr npmDepsHash "$(prefetch-npm-deps ./package-lock.json)"
-nix-build ./release.nix --attr docker --argstr npmDepsHash "$(prefetch-npm-deps ./package-lock.json)"
-nix-build ./release.nix --attr package.linux.x64.elf --argstr npmDepsHash "$(prefetch-npm-deps ./package-lock.json)"
-nix-build ./release.nix --attr package.windows.x64.exe --argstr npmDepsHash "$(prefetch-npm-deps ./package-lock.json)"
-nix-build ./release.nix --attr package.macos.x64.macho --argstr npmDepsHash "$(prefetch-npm-deps ./package-lock.json)"
+# Build the packages for your current platform
+nix build .#polykey-cli
+nix build .#docker
+
+# You can also specify the platform to build for
+nix build .#packages.x86_64-linux.polykey-cli
+nix build .#packages.x86_64-windows.polykey-cli
+nix build .#packages.x86_64-darwin.polykey-cli
```

Comment:
Why change the paths?

```
package.linux.x64.elf
```

Can we just maintain this path for now.

Reply:
Maintain these paths if there's no reason to change the path naming.

Reply:
Okay, there is a reason its like this. When making a flake, you need to specify the system package architecture that will be used to build the flake. Usually, this would be something like `x86_64-linux` or `aarch64-darwin` and so on. These are then used when calling something like nixpkgs.system and related architecture dependent attributes.

At the moment, I have stuck to the recommended naming scheme, as its standardised within the flake ecosystem, but using Nix expressions it is possible for me to map custom system targets to the standardised target naming. So if this is what you would prefer, this can be done.

Reply:
Right so in the case of PK CLI, the only architecture we are using to "build" PK CLI is the `x86_64-linux`.

The target output though may only work on non-NixOS systems, Windows, and Mac.

Like I stated in the other comments, we are not at the point of cross-compiling nodejs itself. It's possible we may never do so if we use Deno and stuff.

I like to think of the JS code as running on a generic JS runtime - that itself should already be precompiled. Don't think of PK CLI as a C project - it is not, one should assume the underlying JS runtime is already precompiled and we just draw it from our dependencies.

That's why the pkg related nix code should actually be moved into our `nixpkgs-matrix` so we can re-use it across different JS-based projects.

Reply:
And if we have deno, that would then a whole another story too. But we're not at the point of thinking about that due to our native binaries atm being non-compatible.

Reply:
Okay, for now I am leaving this as it is not possible to change the paths as easily in a flake output. Flake outputs adhere to a standard for the naming of systems and their derivations, so changing this would deviate from the way flakes are defined. It would probably be easier to leave it as I have done it for now, with the naming of an output such as `packages.x86_64-linux.polykey-cli`, which is automatically translated to the appropriate naming for it to be recognised by `vercel/pkg`. Look at 1d7af25 and the way I have clearly distinguished the build system and then the target systems.